src/services/YotiCallbackProcessor.ts

```diff
@@ -156,9 +157,42 @@ export class YotiCallbackProcessor {
 					  messageCode: MessageCodes.FAILED_TO_WRITE_TXMA,
 				  });
 			  }
-
-			  const { credentialSubject, evidence } =
-				  this.generateVerifiableCredential.getVerifiedCredentialInformation(yotiSessionID, completedYotiSessionInfo, documentFields);
+				
+  			const { given_names, family_name, full_name } = documentFields;
+
+  			const missingGivenName = this.checkMissingField(given_names, "given_names");
+  			const missingFamilyName = this.checkMissingField(family_name, "family_name");
+  			const missingFullName = this.checkMissingField(full_name, "full_name");
+  			let VcNameParts;
+
+  			// If all three name fields are missing, log an error and throw an exception
+  			if (missingGivenName && missingFamilyName && missingFullName) {
+  				this.logger.error({ message: "Missing Name Info in DocumentFields" }, {
+  					messageCode: MessageCodes.VENDOR_SESSION_MISSING_DATA,
+  				});
+  				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing Name Info in DocumentFields");
+  			}
+
+  			if (missingGivenName || missingFamilyName) {
```

Comment:
will be true when both giveName and familyName are null. Is familyName null in Yoti response an expected scenario?
When both are null, should it get the one from incoming VC?

Reply:
if full name is also null then the check above https://github.com/alphagov/di-ipv-cri-f2f-api/blob/F2F-1000/src/services/YotiCallbackProcessor.ts#L171 would log and throw an error as we can't proceed.

Does that answer the question?